cmd/operator/main.go

```diff
@@ -266,11 +267,28 @@ func Main() int {
 		return 1
 	}
 
-	pao, err := prometheusagentcontroller.New(ctx, cfg, log.With(logger, "component", "prometheusagentoperator"), r)
+	var pao *prometheusagentcontroller.Operator
+	verbs := map[string][]string{
+		monitoringv1alpha1.PrometheusAgentName:                           {"get", "list", "watch"},
+		fmt.Sprintf("%s/status", monitoringv1alpha1.PrometheusAgentName): {"update"},
+	}
+
+	cr := k8sutil.NewCRConfig(cfg.Namespaces.AllowList,
+		verbs,
+		monitoringv1alpha1.SchemeGroupVersion.String(),
+		monitoringv1alpha1.PrometheusAgentsKind,
+		monitoringv1alpha1.PrometheusAgentName)
+
+	err = cr.Runnable(ctx, cfg.Host, cfg.TLSInsecure, &cfg.TLSConfig)
 	if err != nil {
-		fmt.Fprint(os.Stderr, "instantiating prometheus-agent controller failed: ", err)
-		cancel()
-		return 1
+		fmt.Fprint(os.Stderr, "pre-requistes for prometheus-agent controller failed: ", err)
```

Comment:
can we use the logger instead?

Reply:
sure I thought so as well. The reason I kept Stderr is for other controllers as well stderr is used. But we can update for them in another PR